src/libstd/sys/unix/backtrace/tracing/gcc_s.rs

```diff
@@ -35,28 +36,40 @@ pub fn write(w: &mut Write) -> io::Result<()> {
     unsafe {
         LOCK.lock();
 
-        writeln!(w, "stack backtrace:")?;
-
-        let mut cx = Context { writer: w, last_error: None, idx: 0 };
-        let ret = match {
-            uw::_Unwind_Backtrace(trace_fn,
-                                  &mut cx as *mut Context as *mut libc::c_void)
-        } {
-            uw::_URC_NO_REASON => {
+        let mut cx = Context {
+            last_error: None,
+            idx: 0,
+            frames: frames,
+        };
+        let result_unwind = uw::_Unwind_Backtrace(trace_fn,
+                                                  &mut cx as *mut Context
+                                                          as *mut libc::c_void);
+        // See libunwind:src/unwind/Backtrace.c for the return values.
+        // No, there is no doc.
+        let ret = match result_unwind {
+            uw::_URC_END_OF_STACK => {
```

Comment:
Sure yeah I can imagine that this function returns a number of values, but we weren't checking any of them before, only for errors. What prompted the change in logic?